src/main/java/com/aws/greengrass/clientdevices/auth/certificate/infra/BackgroundCertificateRefresh.java

```diff
@@ -128,10 +132,39 @@ public void run() {
             return;
         }
 
+        if (!canRun()) {
+            logger.info("Certificates where refreshed recently - retrying later");
+            return;
+        }
+
+        // Why the -10? This is just to avoid a scenario where the scheduler triggers it and the current time
+        // is not exactly after DEFAULT_INTERVAL_SECONDS because this variable gets set a few ms after run is called
+        nextRun = Instant.now().plus(Duration.ofSeconds(DEFAULT_INTERVAL_SECONDS - 10));
```

Comment:
Non-blocking:
This sets a limit on `DEFAULT_INTERVAL_SECONDS`. May be we should document this in a comment beside this default that its value should be set greater than 10. 
(Also I'd like this offset to be as small as possible, considering we're making this tradeoff for few ms. Can we go lower than 10?)

Reply:
I don't really think it matters given it runs once a day and it is a non deterministic value.